crates/bitwarden/src/crypto/asymmetric_crypto_key.rs

```diff
@@ -0,0 +1,99 @@
+use std::str::FromStr;
+
+use base64::Engine;
+use rsa::RsaPrivateKey;
+
+use crate::{
+    crypto::CryptoKey,
+    error::{CryptoError, Error, Result},
+    util::BASE64_ENGINE,
+};
+
+/// An asymmetric encryption key. Used to encrypt and decrypt [`EncString`](crate::crypto::EncString)
+pub struct AsymmetricCryptoKey {
+    pub(in crate::crypto) key: RsaPrivateKey,
+}
```

Comment:
I think we need to add public key here to. Or derive it from the private key.

Usually encrypting is done with the public key, and decrypting using the private.

Reply:
The docs say that RsaPrivateKey implements the PublicKey interface so it should be handled internally by the crate. That said, I haven't implemented RSA encrypt on this PR so haven't had the time to test it.
https://docs.rs/rsa/latest/rsa/struct.RsaPrivateKey.html#method.to_public_key